Source/WebKit/UIProcess/Extensions/WebExtension.h

```diff
@@ -240,11 +245,19 @@ class WebExtension : public API::ObjectImpl<API::Object::Type::WebExtension>, pu
     NSString *sidebarTitle();
 #endif
 
-    CocoaImage *imageForPath(NSString *, NSError **);
+    CocoaImage *imageForPath(NSString *, NSError **, CGSize sizeForResizing = CGSizeZero);
 
+    size_t bestSizeInIconsDictionary(NSDictionary *, size_t idealPixelSize);
     NSString *pathForBestImageInIconsDictionary(NSDictionary *, size_t idealPixelSize);
-    CocoaImage *bestImageInIconsDictionary(NSDictionary *, size_t idealPointSize, NSError **);
-    CocoaImage *bestImageForIconsDictionaryManifestKey(NSDictionary *, NSString *manifestKey, CGSize idealSize, RetainPtr<CocoaImage>& cacheLocation, Error, NSString *customLocalizedDescription);
+
+    CocoaImage *bestImageInIconsDictionary(NSDictionary *, CGSize idealSize, const Function<void(NSError *)>&);
```

Comment:
Just out of curiosity -- what was the reason behind using a callback to report errors from this method? Could we theoretically also use a `wtf::Result` here?

Reply:
Yes `Expected` is an option I considered, but the error needs to be reported to the context or the extension depending on the client calling it. The callback cleaned up the call sites a lot.

Reply:
Oh and the main reason I needed a lambda was because multiple errors can be reported now, since we load different image paths and if one ore more fail the lambda will be called for each.